### ISSUE TITLE
Add pv.kubernetes.io/migrated-to to deploy defaults for PersistentVolume and PersistentVolumeClaim

### ISSUE THREAD

**Issue by voor** [k14s/kapp#605]
**Describe the problem/challenge you have**
Upgrading to Kubernetes 1.23 requires migrating off the in-tree storage classes, this means any clusters that are migrating are getting  the annotation `pv.kubernetes.io/migrated-to` applied to their `PersistentVolumeClaim`.  There is already a rule for:
```
- paths:
  - [metadata, annotations, pv.kubernetes.io/bind-completed]
  - [metadata, annotations, pv.kubernetes.io/bound-by-controller]
  - [metadata, annotations, volume.beta.kubernetes.io/storage-provisioner]
  - [spec, storageClassName]
  - [spec, volumeMode]
  - [spec, volumeName]
  type: copy
  sources: [new, existing]
  resourceMatchers:
  - apiVersionKindMatcher: {apiVersion: v1, kind: PersistentVolumeClaim}
```

**Describe the solution you'd like**
Would like to add:
```
  - [metadata, annotations, pv.kubernetes.io/migrated-to]
```

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by praveenrewar** [comment]
Sounds reasonable to me :)

**Comment by renuy** [comment]
Released as part of v0.53.0
